### ISSUE TITLE
[Feature Reqeust] Add force_cast(null as integer) support

### ISSUE THREAD

**Issue by diyism** [crate/crate#4514]
**Feature description**:
Now i use the following line to increase null field:
update statis_user_success set cnt=cast(concat('0', cnt) as integer)+1 where key='898234';

Hope:
update statis_user_success set cnt=force_cast(cnt as integer)+1 where key='898234';

**Comment by mfussenegger** [comment]
Thx for the suggestion.

For your use case there is the [coalesce](https://crate.io/docs/reference/sql/scalar.html?highlight=coalesce#coalesce-first-arg-second-arg-returns-same-type-as-arguments) function:

```
update statis_user_success set cnt = coalesce(cnt, 0) + 1 where key = '898234';
```

I don't see any advantage that a `force_cast` function would bring, other than that it might be less to type in certain cases. But it's less flexible than adding a coalesce call and also requires an implicit definition of a default value per type.

**Comment by diyism** [comment]
I'm using CrateDB Version: 0.56.3,
when "select coalesce(null, 0)+1 from statis_user_success limit 1;",  result: 1,
when "select cnt from statis_user_success limit 1;" result: NULL
when "select coalesce(cnt, 0)+1 from statis_user_success limit 1;",   result: SQLActionException[UnsupportedFeatureException: all arguments for coalesce function must have the same data type]

**Comment by mfussenegger** [comment]
This happens because `cnt` is an integer and `0` is interpreted as a long.

`coalesce` currently does not support mixed arguments `(int, long)`.

See:

```
cr> select cnt, coalesce(cnt, 0) + 1 as c from statis_user_success limit 1;
SQLActionException[UnsupportedFeatureException: all arguments for coalesce function must have the same data type]
cr> select cnt, coalesce(cnt, cast(0 as integer)) + 1 as c from statis_user_success limit 1;
+------+---+
|  cnt | c |
+------+---+
| NULL | 1 |
+------+---+
```

We'll improve this behaviour eventually, so that the first example works as well

**Comment by matriv** [comment]
Feature is included in [2.3.0](https://crate.io/docs/crate/reference/en/latest/appendices/release-notes/2.3.0.html)
